### DISCUSSION TITLE
Minimize HTML / Cluster atomic CSS classes into single CSS classes in the production build

### DISCUSSION THREAD

**Discussion by deleonio** [unocss/unocss#481] [Ideas]
Would it be a cool feature if in a production build all the atomic CSS classes were neatly clustered, to become slim HTML?

I think the algorithm would be expensive.

Code (Development)
```html
<div class="w-full bg-blue h-full">
</div>
<div class="w-full bg-blue font-sans">
</div>
```

Code (Production)
```css
uno-23ioejfi {
  @apply w-full bg-blue
}
```

```html
<div class="uno-3iefi h-full">
</div>
<div class="uno-3iefi font-sans">
</div>
```

**Comment by antfu** [comment]
This involves with very complex source code rewrite and it's very hard to do it right. (Windi has a complication mode, which the vite plugin does not support it either). This is out-of-scope of this project, but feel free to experiment if you figure out some ways to do it.

**Comment by vipero07** [comment]
I like this idea, but tooling does exist if it isn't added. https://www.npmjs.com/package/clean-css#what-level-2-optimizations-do
